index.js

```diff
@@ -40,13 +43,15 @@ const pDebounce = (fn, wait, options = {}) => {
 pDebounce.promise = function_ => {
 	let currentPromise;
 
-	return async (...arguments_) => {
+	// Also see notes in pDebounce() about preserving calling context
+	return async function (...arguments_) {
 		if (currentPromise) {
 			return currentPromise;
 		}
 
+		const context = this;
 		try {
-			currentPromise = function_(...arguments_);
+			currentPromise = function_.apply(context, arguments_);
```

Comment:
```suggestion
			currentPromise = function_.apply(this, arguments_);
```

Reply:
Sure - I made this change.  I understand you aren't interested in a change that makes the code better for transpilation. https://github.com/sindresorhus/p-throttle/pull/25#discussion_r561542921